lib/Reader/readerir.cpp

```diff
@@ -1411,8 +1411,14 @@ Type *GenIR::getClassType(CORINFO_CLASS_HANDLE ClassHandle, bool IsRefClass,
     for (uint32_t I = 0; I < NumDerivedFields; I++) {
       CORINFO_FIELD_HANDLE FieldHandle = getFieldInClass(ClassHandle, I);
       if (FieldHandle == nullptr) {
-        // Likely a class that derives from System.__ComObject. See
-        // LLILC issue #557.
+        if (IsRefClass) {
+          // Likely a class that derives from System.__ComObject. See
+          // LLILC issue #557. We'll just have to cope with an incomplete
+          // picture of this type.
+          break;
+        }
+        // For value classes we use the LLVM type to derive the GC reporting.
+        // So we can't tolerate any missing fields.
         throw NotYetImplementedException("can't account for all fields");
```

Comment:
Should this be an assert instead of an NYI? We don't know of any existing value classes that have this issue, right?